client/src/hooks/auth.hook.js

```diff
@@ -0,0 +1,32 @@
+import { useState, useCallback, useEffect } from 'react';
+
+const storageName = 'userData';
+
+export const useAuth = () => {
+  const [token, setToken] = useState(null);
+  const [userId, setUserId] = useState(null);
+
+  const login = useCallback((jwtToken, id) => {
+    setToken(jwtToken);
+    setUserId(id);
+    localStorage.setItem(
+      storageName,
+      JSON.stringify({ userId: id, token: jwtToken })
+    );
+  }, []);
+
+  const logout = useCallback(() => {
+    setToken(null);
+    setUserId(null);
+    localStorage.removeItem(storageName);
+  }, []);
+
+  useEffect(() => {
+    const data = JSON.parse(localStorage.getItem(storageName));
```

Comment:
Make sure to catch errors here. What happens when localStorage is empty or the value is not a JSON